docs/website/docs/general-usage/credentials/config_specs.md

```diff
@@ -0,0 +1,327 @@
+
+Usage:
+```python
+credentials = ConnectionStringCredentials()
+
+# Set the necessary attributes
+credentials.drivername = "postgresql"
+credentials.database = "my_database"
+credentials.username = "my_user"
+credentials.password = "my_password"
+credentials.host = "localhost"
+credentials.port = 5432
+
+# Convert credentials to connection string
+connection_string = credentials.to_native_representation()
+
+# Parse a connection string and update credentials
+native_value = "postgresql://my_user:my_password@localhost:5432/my_database"
+credentials.parse_native_representation(native_value)
+
+# Get a URL representation of the connection
+url_representation = credentials.to_url()
+```
+
+### OAuth2Credentials
+
+The OAuth2Credentials class handles OAuth 2.0 credentials, including client ID,
+client secret, refresh token, and access token.
+It also allows for the addition of scopes and provides methods for client authentication.
+
+Usage:
+```python
+credentials = OAuth2Credentials(
+    client_id="CLIENT_ID",
+    client_secret="CLIENT_SECRET",
+    refresh_token="REFRESH_TOKEN",
+    scopes=["scope1", "scope2"]
+)
+
+# Authorize the client
+credentials.auth()
+
+# Add additional scopes
+credentials.add_scopes(["scope3", "scope4"])
+```
+
+### GCP Credentials
+
+#### GcpServiceAccountCredentials
+
+The GcpServiceAccountCredentials class manages GCP Service Account credentials.
+It can parse native values either as ServiceAccountCredentials or as serialized `services.json`.
+This class provides methods to retrieve native credentials for Google clients.
+
+Usage:
+```python
+credentials = GcpServiceAccountCredentials()
+
+# Parse a native value (ServiceAccountCredentials)
+# Accepts a native value, which can be either an instance of ServiceAccountCredentials
+# or a serialized services.json.
+# Parses the native value and updates the credentials.
+native_value = ...
+credentials.parse_native_representation(native_value)
+
+# Retrieve native credentials for Google clients
+# For example, build the service object for Google Analytics PI.
+client = BetaAnalyticsDataClient(credentials=credentials.to_native_credentials())
+
+# Get a string representation of the credentials
+# Returns a string representation of the credentials in the format client_email@project_id.
+credentials_str = str(credentials)
+```
+
+#### GcpOAuthCredentials
+
+The GcpOAuthCredentials class is responsible for handling OAuth2 credentials for
+desktop applications in Google Cloud Platform (GCP).
+It can parse native values either as GoogleOAuth2Credentials or as
+serialized OAuth client secrets JSON.
+This class provides methods for authentication and obtaining access tokens.
+
+Usage:
+```python
+oauth_credentials = GcpOAuthCredentials()
+
+# Accepts a native value, which can be either an instance of GoogleOAuth2Credentials
+# or serialized OAuth client secrets JSON.
+# Parses the native value and updates the credentials.
+native_value_oauth = ...
```

Comment:
I'd also wrap example in a dlt.source.
the same remarks regarding default credentials. you can mention that it works in Colab

we need a separate paragraph on authentication process.
In order for `auth()` method to succeeed:
- you must provide valid `client_id` and `client_secret`, `refresh_token` and `token_uri` in order to get a current access token and authenticate with oauth. mind that the `refresh_token` must contain all the scopes that you require for your access.
- if refresh token is not provided and you run the pipeline from a console or a notebook, `dlt` will use `InstalledAppFlow` to run the desktop authentication flow.

please also link to one of our source docs where we show how to configure oauth for Google. the link to setup script below should staty - spot on!